tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Builders/OdsControllerBuilder.cs

```diff
@@ -8,23 +8,37 @@ namespace NHSD.BuyingCatalogue.Organisations.Api.UnitTests.Builders
     internal sealed class OdsControllerBuilder
     {
         private IOdsRepository odsRepository;
+        private IOrganisationRepository orgRepository;
 
         private OdsControllerBuilder()
         {
             odsRepository = Mock.Of<IOdsRepository>();
+            orgRepository = Mock.Of<IOrganisationRepository>();
         }
 
         internal static OdsControllerBuilder Create()
         {
             return new();
         }
 
-        internal OdsControllerBuilder WithGetByOdsCode(OdsOrganisation result)
+        internal OdsControllerBuilder WithGetByOdsCode(OdsOrganisation result, Organisation organisation)
         {
             var odsRepositoryMock = new Mock<IOdsRepository>();
-            odsRepositoryMock.Setup(r => r.GetBuyerOrganisationByOdsCodeAsync(It.IsAny<string>())).ReturnsAsync(result);
-
+            if (result != null)
+            {
+                odsRepositoryMock.Setup(r => r.GetBuyerOrganisationByOdsCodeAsync(result.OdsCode)).ReturnsAsync(result);
+            }
             odsRepository = odsRepositoryMock.Object;
```

Comment:
Linting error – should be a line break after the closing brace.

Reply:
updated